### DISCUSSION TITLE
ATV

### DISCUSSION THREAD

**Discussion by berkayakciger** [iptv-org/community#952] [Q&A]
### Channel ID

ATV.tr

### Stream URL (optional)

https://www.atv.com.tr/canli-yayin

### Notes (optional)

ATV

### Please confirm the following

- [X] I have read [Contributing Guide](https://github.com/iptv-org/iptv/blob/master/CONTRIBUTING.md#request-a-channel)

**Comment by BellezaEmporium** [comment]
Tokenized, needs a specific manipulation in order to get the tokened stream.

**Comment by arecki2** [comment]
Oh, nice that you've touched the tokenization issue 🙂 Lots of good channels had gone into tokenizing their streams (personally I used to enjoy Rai News 24 but they tokenized themselves too), but in many cases you can still retrieve your URL, play it in VLC or other player, and it may even remain valid for some time, even if you stop the playback in the meantime.

I know Streamlink supports some of these streams, e.g. A Haber, but in order to add them to these playlists, we should have that Streamlink running on a server... perhaps even a personal copy thereof for each user?

Or can it be done the same way ythls.onrender.com works?

**Comment by BellezaEmporium** [comment]
> Oh, nice that you've touched the tokenization issue 🙂 Lots of good channels had gone into tokenizing their streams (personally I used to enjoy Rai News 24 but they tokenized themselves too), but in many cases you can still retrieve your URL, play it in VLC or other player, and it may even remain valid for some time, even if you stop the playback in the meantime.
> 
> I know Streamlink supports some of these streams, e.g. A Haber, but in order to add them to these playlists, we should have that Streamlink running on a server... perhaps even a personal copy thereof for each user?
> 
> Or can it be done the same way ythls.onrender.com works?

That's what my project https://github.com/BellezaEmporium/query-streamlink and https://github.com/back-to/liveproxy (from one Streamlink member) does somehow someway.

**Comment by freearhey** [comment]
https://github.com/iptv-org/iptv/blob/aec1bdb0a7037bf6f7112d2be3bdd93a50d5209f/streams/tr.m3u#L79-L88
